src/clib/pio_darray.c

```diff
@@ -1357,32 +1359,37 @@ int flush_output_buffer(file_desc_t *file, bool force, PIO_Offset addsize)
     maxusage = usage;
   }
   if(force || usage>=PIO_BUFFER_SIZE_LIMIT){
-    int request[2*PIO_MAX_VARS];
-    int status[2*PIO_MAX_VARS];
-    int rcnt=0;
+    int rcnt;
     bool prev_dist=false;
     int prev_record=-1;
     int prev_type=0;
-
+    int  maxreq;
+    int reqcnt;
+    maxreq = 0;
+    reqcnt=0;
+    rcnt=0;
     for(int i=0; i<PIO_MAX_VARS; i++){
       vdesc = file->varlist+i;
+      reqcnt+=vdesc->nreqs;
+      if(vdesc->nreqs>0) maxreq = i;
+    }
+    int request[reqcnt];
```

Comment:
Wow I didn't even know you could do that in C. I would have thought that a malloc was required. Didn't know you could declare an array with a size you just computed!